src/com/intellij/idea/plugin/hybris/project/descriptors/YModuleLibDescriptorUtil.kt

```diff
@@ -63,6 +65,9 @@ object YModuleLibDescriptorUtil {
             val libraryModel = modifiableModelsProvider.getModifiableLibraryModel(library)
             if (addJarDirectory) libraryModel.addJarDirectory(VfsUtil.getUrlForLibraryRoot(libraryDirRoot), true)
             else libraryModel.addRoot(VfsUtil.getUrlForLibraryRoot(libraryDirRoot), OrderRootType.CLASSES)
+            if (sourcesDirRoot != null && ApplicationSettingsComponent.getInstance().state.withStandardProvidedSources) {
```

Comment:
looks like something is missing, there is a method `getStandardSourceJarDirectory` which has to register corresponding sources.

can you please investigate how can we rely on `DOC_SOURCES_PARENT_JAR_PATH` constant.

Reply:
`getStandardSourceJarDirectory` is not being called for backoffice extensions and I couldn't get that working for backoffice libraries from there.

Method `getStandardSourceJarDirectory` registers `hybris/bin/modules/backoffice-framework/backoffice/doc/sources` directory only for `backoffice` root extension which is not enough.

Reply:
```bash
find . -name sources                                               
./hybris/bin/platform/ext/europe1/doc/sources
./hybris/bin/platform/ext/hac/doc/sources
./hybris/bin/modules/backoffice-framework/backoffice/doc/sources`
```

FYI. out of the box SAP provide only three directories with name sources, and only the backoffice one has meaningful content :)

Reply:
interesting, so far I see one alternative way to do it via creation and customisation of the `JavaLibraryDescriptor` when we request creation of new libs in the `YModuleLibDescriptorUtil`, yet THAT alternative will require significant refactoring.

Reply:
interesting, so far I see one alternative way to do it via creation and customisation of the `JavaLibraryDescriptor` when we request creation of new libs in the `YModuleLibDescriptorUtil`, yet THAT alternative will require significant refactoring.